exporter/awsxrayexporter/awsxray.go

```diff
@@ -137,3 +150,38 @@ func wrapErrorIfBadRequest(err error) error {
 	}
 	return err
 }
+
+// encodeOtlpAsBase64 builds bytes from traces and generate base64 value for them
+func encodeOtlpAsBase64(td ptrace.Traces, cfg *Config) ([]*string, error) {
+	var documents []*string
+	marshaller := &ptrace.ProtoMarshaler{}
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		// 1. build a new trace with one resource span
+		singleTrace := ptrace.NewTraces()
+		td.ResourceSpans().At(i).CopyTo(singleTrace.ResourceSpans().AppendEmpty())
```

Comment:
Have to copy to a new trace?

Reply:
because I didn't find a function that takes resourcespans as the input and produce bytes. The only function i found is this `MarshalTraces`, which takes trace as the input.

Reply:
I think you need to put them under traces, this is the format used by the export request protobuf.
Basically the protobuf schema has same structure for `TracesDatat` and `ExportTraceServiceRequest`

- https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L38
- https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/collector/trace/v1/trace_service.proto#L36

Reply:
Can you try using `ExportTraceServiceRequest` https://opentelemetry.io/docs/specs/otlp/#binary-protobuf-encoding I think it should work as well and is closer to otlp implementation, which may have some other information in the future.

Reply:
ExportTraceServiceRequest is the `Orig` of `Trace`, however `Orig` is not being exposed publicly. there's a function as below, but as you can see, it's a private function.
```
func (ms Traces) getOrig() *otlpcollectortrace.ExportTraceServiceRequest {
	return internal.GetOrigTraces(internal.Traces(ms))
}
```